### ISSUE TITLE
can't really disable logs

### ISSUE THREAD

**Issue by michaelpecan** [SergeyPirogov/webdriver_manager#209]
I have a logger class that inherits from logger python and overrides the _log() method, webdriver manager tries to use it and fails, I would like to disable logs outside of webdriver manager log method cause the current disable functionality is to disable log parameter WDM_LOG_LEVEL on a logger that it's _log method is different.

**Comment by DanielOberlechner** [comment]
Hi,
I'm not quite sure but I think I have had the same problem:
I tried the followning:
`driver = webdriver.Chrome(ChromeDriverManager(log_level=0, print_first_line=False).install(), options=chrome_options)`
This did not deactivate logging, that worked only if I did set the environment variable like this:
`os.environ['WDM_LOG_LEVEL'] = '0'`
I wasn't quite sure if only I'm an idiot and did something wrong, because I'm not an advanced programmer,
but it seems someone else has the same problem like me so I think it is something real ...

Any desire to think this problem?

Kind regards,
by Daniel :)

**Comment by batu1579** [comment]
hi,

I'm sorry i have to write in Chinese, because my English is really poor and I can't express myself clearly, so I have to ash you to use the Google Translate.

我看了一下源码，发现在所有浏览器的基类 `DriverManager` （在 `manager.py` 中）中并没有使用过传入的 `log_level` 参数，所以你在调用浏览器驱动管理器类似于 `ChromeDriverManager` 时传入这个参数是没有用的。在浏览源码的过程中我发现整个包中的日志输出都是靠`logger.py` 文件中的 `log` 函数实现的。具体代码如下：

```python
loggers = {}


def log(text, level=logging.INFO, name="WDM", first_line=False, formatter='[%(name)s] - %(message)s'):
    log_level = os.getenv('WDM_LOG_LEVEL')
    if log_level:
        level = int(log_level)
    if loggers.get(name):
        loggers.get(name).info(text)
    else:
        _logger = logging.getLogger(name)

        handler = logging.StreamHandler()
        formatter = logging.Formatter(formatter)
        handler.setFormatter(formatter)
        _logger.addHandler(handler)
        _logger.setLevel(level)
        loggers[name] = _logger
        _logger.info(text)
```



作者貌似是用 `name` 参数来给 `logging` 模块中的 `logger` 分组，将声明过的 `logger`都会存储在 `logger.py` 文件中名叫 `loggers` 的字典中，以此来避免每次输出日志都要重新声明一个 `logger` 。可以看到所有的日志全都是 `info` 级别的， 同时在声明的过程中作者是使用 `level` 参数来调整这个 `logger` 的过滤等级的。 我看大多数时候作者使用的都是 `name` 参数的默认值，那么我们也许可以用下面的方法在不改变包中代码的同时禁用输出日志：

在整个程序的最开头先引入包中负责日志记录的 `log` 函数：

```python
from webdriver_manager.logger import log
```

然后立刻使用 `log` 函数创建一个指定等级的 `logger` :

```
log("", level=50)
```

因为参数 `level` 指定的是 `logger` 的记录等级所以想禁用日志可以让 `logger` 只记录最高等级日志既 `critical` 级别，同时因为包中所有的日志都只使用 `info` 级别，于是包中的所有日志就都不会输出了！

另外我觉得作者应该是有自己的计划，只不过还没有编写完成，通过预留的 `logger` 组名称就可以看出（当然也可能是我想多了，这只是作者的习惯写法）。不过在作者完善这个功能之前这个方法应该可以暂时达到禁用日志的效果。

batu1579

**Comment by batu1579** [comment]
> Hi,
> I'm not quite sure but I think I have had the same problem:
> I tried the followning:
> `driver = webdriver.Chrome(ChromeDriverManager(log_level=0, print_first_line=False).install(), options=chrome_options)`
> This did not deactivate logging, that worked only if I did set the environment variable like this:
> `os.environ['WDM_LOG_LEVEL'] = '0'`
> I wasn't quite sure if only I'm an idiot and did something wrong, because I'm not an advanced programmer,
> but it seems someone else has the same problem like me so I think it is something real ...
> 
> Any desire to think this problem?
> 
> Kind regards,
> by Daniel :)

Maybe you can try my way of disabling logging. hope it helps.

**Comment by Alyetama** [comment]
If none of the options to disable logs works, you can try this simple workaround:

```python
import os, sys
stdout = sys.stdout
with open(os.devnull, 'w') as f:
    sys.stdout = f
    driver = webdriver.Chrome(ChromeDriverManager().install(), options=chrome_options)
sys.stdout = stdout
```

**Comment by dariopnc** [comment]
This should be fixed by #170 , or at least give you a way to completely disable the logging by controlling directly the logger. The current master branch on here feature this merge, but this version hasn't yet been tagged to relaease and published to pip

**Comment by Unconventional23** [comment]
Sorry I don't think the issue has been fixed yet even on the version 3.5.0
Both the solutions doesn't seem be working,
```
ChromeDriverManager("2.26", print_first_line=False).install()
os.environ['WDM_PRINT_FIRST_LINE'] = 'False'
```
Or is there something I am missing out?

**Comment by aleksandr-kotlyar** [comment]
For some users it works, for some is not. Depends on log configuration in the project.
PR #245 should honestly resolve this issue.
